src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -586,55 +736,76 @@ export class SavedObjectsRepository {
       return { saved_objects: [] };
     }
 
-    const unsupportedTypeObjects = objects
-      .filter(o => !this._allowedTypes.includes(o.type))
-      .map(({ type, id }) => {
-        return ({
-          id,
-          type,
-          error: SavedObjectsErrorHelpers.createUnsupportedTypeError(type).output.payload,
-        } as any) as SavedObject<T>;
-      });
+    let bulkGetRequestIndexCounter = 0;
+    const bulkGetDocs: object[] = [];
+    const expectedBulkGetResults: Array<Either<any, any>> = objects.map(object => {
```

Comment:
I don't think we can do that, because the second map parameter is equivalent to the current index in the `object` array. If we do a bulkGet with a disallowed type, the resulting cluster call will not include it. So the `bulkGetRequestIndexCounter` can wind up being smaller than `objects.length`.